### PR TITLE
fix: should remove mockPath from callstack whether success or failed

### PR DESCRIPTION
**Pull request by miserylee**
### Description

When mock a module import exception, the mock function throws but the mockPath not removed from the callstack  
this makes the retry import  not perform correctly

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Comment by sheremet-va** [comment]
Looks good, but can you add a test case where this fails without your change? You can add it to examples/mocks.

File: examples/mocks/src/retry-dynamic-import.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export async function retryDynamicImport() {
+  let retryTimes = 0
+  const load = async () => {
+    try { // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-expect-error
```
Comment (sheremet-va) [inline examples/mocks/src/retry-dynamic-import.ts:5]: Why do you need this here?

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: examples/mocks/src/retry-dynamic-import.ts (hunk 1 above)
Comment (miserylee) [reply examples/mocks/src/retry-dynamic-import.ts:5]: This case is for dynamic import in browser, bundled by webpack.  
The modules dynamically imported are from remote over network, so it maybe failed cause network exception or any unknown reason.  
We want to retry import several times for reducing the final failed possibility.

Comment (miserylee) [reply examples/mocks/src/retry-dynamic-import.ts:5]: Emmm , did you mean the comments upon?  
Without the comments, the ts language service will emit an error diagnostic message as below 
image

Comment (miserylee) [reply examples/mocks/src/retry-dynamic-import.ts]: I have appended a commit, changed the import path to have extension, to avoid this ts check issue.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**